### PR TITLE
Add space and subnet to schema

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Add space and subnet to schema

Also add `IF NOT EXISTS` to indexes and `provider` table for relations

### CODE REVIEW

**Review by manadart — CHANGES_REQUESTED** [review]
We don't need the provider table. PROVIDER_ID is the provider's ID for the entity. We currently have a collection providerIDs, which tracks these across different entity types, but I'd leave that for now. Just put a unique index on it.

**Comment by jack-w-shaw** [comment]
Fixed, ready for another look over

File: schema.ddl [hunk 1]
```diff
@@ -27,5 +30,46 @@ CREATE TABLE IF NOT EXISTS machine_filesystem (
...
+	is_public			BOOLEAN,
+	space_id			TEXT NOT NULL,
+	fan_local_underlay	TEXT,
+	fan_overlay			TEXT,
+	CONSTRAINT			fk_subnet_space
+		FOREIGN KEY		(space_id)
+		REFERENCES		space(id)
+);
+
+CREATE UNIQUE INDEX IF NOT EXISTS idx_space_provider_id
```
Comment (manadart) [inline schema.ddl]: Should be for subnet provider ID.

**Review by manadart — CHANGES_REQUESTED** [review]
One nit.

File: schema.ddl (hunk 1 above)
Comment (jack-w-shaw) [reply schema.ddl]: resolved

**Review by manadart — APPROVED** [review]

**Pull request merged**